### PR TITLE
8344667: Remove most uses of AWT Permissions from the desktop module

### PR DESCRIPTION
**Pull request by prrace**
This removes most uses of AWTPermission in the desktop module.
clipboard will be handled separately because it is less straightforward.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344667](https://bugs.openjdk.org/browse/JDK-8344667): Remove most uses of AWT Permissions from the desktop module (**Bug** - P4)

### Reviewers
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)
 * [Alexander Zuev](https://openjdk.org/census#kizune) (@azuev-java - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22350/head:pull/22350` \
`$ git checkout pull/22350`

Update a local copy of the PR: \
`$ git checkout pull/22350` \
`$ git pull https://git.openjdk.org/jdk.git pull/22350/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22350`

View PR using the GUI difftool: \
`$ git pr show -t 22350`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22350.diff">https://git.openjdk.org/jdk/pull/22350.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22350#issuecomment-2496388784)
</details>

### CODE REVIEW

**Review by azvegint — COMMENTED** [review]
A few places that probably need some attention:

`./src/java.desktop/share/classes/java/awt/Composite.java:60: * to a screen device is governed by the {@code readDisplayPixels}`

```
 * Since this interface must expose the contents of pixels on the
 * target device or image to potentially arbitrary code, the use of
 * custom objects which implement this interface when rendering directly
 * to a screen device is governed by the {@code readDisplayPixels}
 * {@link AWTPermission}.  The permission check will occur when such
 * a custom object is passed to the {@code setComposite} method
 * of a {@code Graphics2D} retrieved from a {@link Component}.
```
------

doc update:

```
`./src/java.desktop/share/classes/java/awt/doc-files/Modality.html:338:      A special <code>AWTPermission</code>, <code>"toolkitModality"</code>,`

    <p>
      A special <code>AWTPermission</code>, <code>"toolkitModality"</code>,
      is required to show toolkit-modal
      dialogs. This would prevent, for example, blocking a browser or
      Java Web Start (JWS) by modal dialogs shown from applets.
    </p><p>
      The same permission is required to exclude a window from toolkit modality.
      This would prevent, for example, a dialog shown from an applet not to be
      blocked by a browser's or JWS's modal dialog.

```

---

```
./test/jdk/lib/client/ExtendedRobot.java:71:     * @throws  SecurityException if {@code createRobot} permission is not granted
./test/jdk/lib/client/ExtendedRobot.java:75:     * @see     java.awt.AWTPermission
./test/jdk/lib/client/ExtendedRobot.java:102:     * @throws  SecurityException if {@code createRobot} permission is not granted
./test/jdk/lib/client/ExtendedRobot.java:107:     * @see     java.awt.AWTPermission
```

-----

File: src/java.desktop/share/classes/javax/swing/SwingUtilities.java [hunk 1]
```diff
@@ -1970,14 +1970,6 @@ public void windowDeactivated(WindowEvent e) {
         public void show() {
             // This frame can never be shown
         }
-        public void dispose() {
```
[the remark this answers is not in the document]
Comment (prrace) [reply src/java.desktop/share/classes/javax/swing/SwingUtilities.java:1973]: Yes I noticed these too, and yesterday I filed a bug to track that https://bugs.openjdk.org/browse/JDK-8344918

**Comment by prrace** [comment]
> A few places that probably need some attention:
> 
> `./src/java.desktop/share/classes/java/awt/Composite.java:60: * to a screen device is governed by the {@code readDisplayPixels}`

If you take a look at https://github.com/openjdk/jdk/pull/22338/files you will see this text is being addressed there.


> 
> doc update:
> 
> ```
> `./src/java.desktop/share/classes/java/awt/doc-files/Modality.html:338:      A special <code>AWTPermission</code>, <code>"toolkitModality"</code>,`
> 
>     <p>
>       A special <code>AWTPermission</code>, <code>"toolkitModality"</code>,
>       is required to show toolkit-modal
>       dialogs. This would prevent, for example, blocking a browser or
>       Java Web Start (JWS) by modal dialogs shown from applets.
>     </p><p>
>       The same permission is required to exclude a window from toolkit modality.
>       This would prevent, for example, a dialog shown from an applet not to be
>       blocked by a browser's or JWS's modal dialog.
> ```


thanks for reminding me about these doc-files. I've been meaning to check them.
Since those files are likely specification  I'll add that to the upcoming fix to deprecate AWTPermission since I can bundle into the CSR for that. not here.


>  ```
> ./test/jdk/lib/client/ExtendedRobot.java:71:     * @throws  SecurityException if {@code createRobot} permission is not granted
> ./test/jdk/lib/client/ExtendedRobot.java:75:     * @see     java.awt.AWTPermission
> ./test/jdk/lib/client/ExtendedRobot.java:102:     * @throws  SecurityException if {@code createRobot} permission is not granted
> ./test/jdk/lib/client/ExtendedRobot.java:107:     * @see     java.awt.AWTPermission
> ```

I guess I could do that. No spec. issue around that.

**Review by azvegint — APPROVED** [review]

**Review by azuev-java — APPROVED** [review]

**Pull request closed without merging**